### PR TITLE
372 rest refactor phase2

### PR DESCRIPTION
**Pull request by selimnairb**
Addresses #372,

### CODE REVIEW

File: hs_core/urls.py [hunk 1]
```diff
@@ -4,11 +4,11 @@
 urlpatterns = patterns('',
 
     # resource API
-    url(r'^resource/(?P<pk>[A-z0-9]+)/$', views.resource_rest_api.ResourceListRetrieveCreateUpdateDelete.as_view(),
-        name='get_update_delete_resource'),
+    url(r'^resourceList/$', views.resource_rest_api.ResourceList.as_view(),
```
Comment (ironfroggy) [inline hs_core/urls.py]: why the change in URLs? What was wrong with `resource/`and `resource/ID/`?

File: hs_core/views/resource_rest_api.py [hunk 2]
```diff
@@ -163,34 +114,88 @@ def get_queryset(self):
             res = res.union(resources)
 
         for r in res:
-            resource_list_item = self._resourceToResourceListItem(r)
+            resource_list_item = _resourceToResourceListItem(r)
```
Comment (ironfroggy) [inline hs_core/views/resource_rest_api.py]: If this is shared between classes should there be a common subclass it is part of?

File: hs_core/urls.py (hunk 1 above)
Comment (selimnairb) [inline hs_core/urls.py]: resourceList is in the spec, so I'm following the spec.

**Comment by selimnairb** [comment]
@ironfroggy Good point.  I made the shared function into a mixin.

**Comment by selimnairb** [comment]
Unless @pkdash or @horsburgh have any objections, I think this branch can be merged.
